companyweb_base/models/partner_cweb_model.py

```diff
@@ -0,0 +1,527 @@
+                self.cweb_image_enable = False
+        else:
+            self.cweb_score = None
+            self.cweb_image = None
+            self.cweb_image_enable = False
+
+    def _populate_cweb_data(self, cweb_response):
+
+        self.cweb_creditLimit_enable = cweb_response["CreditLimit"]["IsEnabled"]
+        cweb_has_creditLimit_values = cweb_response["CreditLimit"]["Value"]
+        if self.cweb_creditLimit_enable and cweb_has_creditLimit_values:
+            limit = cweb_response["CreditLimit"]["Value"]["Limit"]
+            if limit or limit == 0:
+                self.cweb_creditLimit = limit
+                self.cweb_creditLimit_enable = True
+            else:
+                self.cweb_creditLimit_enable = False
+            if cweb_response["CreditLimit"]["Value"]["Info"]:
+                self.cweb_creditLimit_info = cweb_response["CreditLimit"]["Value"][
+                    "Info"
+                ]
+                self.cweb_creditLimit_info_enable = True
+            else:
+                self.cweb_creditLimit_info_enable = False
+        else:
+            self.cweb_creditLimit = None
+            self.cweb_creditLimit_info = None
+            self.cweb_creditLimit_info_enable = False
+
+        self.cweb_warnings_enable = cweb_response["WarningsOverview"]["IsEnabled"]
+        cweb_has_warnings_values = cweb_response["WarningsOverview"]["Value"]
+        separator = "\n"
+        if self.cweb_warnings_enable and cweb_has_warnings_values:
+            if cweb_response["WarningsOverview"]["Value"]["Warnings"]:
+                self.cweb_warnings = separator.join(
+                    cweb_response["WarningsOverview"]["Value"]["Warnings"]["string"]
+                )
+            else:
+                self.cweb_warnings_enable = False
+        else:
+            self.cweb_warnings = None
+
+    def button_companyweb_enhance(self):
+        """Main logic of the module
+        Validate that the logged in user has credentials
+        make the API CALL
+        based on status make decision
+        When status is ok -> populate fields"""
+        user_login = self.env.user.cweb_login
+        user_password = self.env.user.cweb_password
+        if not user_login or not user_password:
+            return self._call_wizard_credentials("No Companyweb credentials")
+
+        client = zeep.Client("https://connect.companyweb.be/V1.3/alacarteservice.asmx")
+        r = client.service.GetCompanyByVat(
+            dict(
+                CompanyWebLogin=user_login,
+                CompanyWebPassword=user_password,
+                ServiceIntegrator=SERVICE_INTEGRATOR_ID,
+                LoginHash=self._create_hash(
+                    user_login, user_password, SERVICE_INTEGRATOR_SECRET
+                ),
+                Language="FR",
+                VatNumber=self.vat,
+            )
+        )
+        if r["StatusCode"] in [101, 302]:
+            return self._call_wizard_credentials("Bad Companyweb credentials")
```

Comment:
```suggestion
            return self._call_wizard_credentials("Enter Companyweb credentials")
```

Reply:
We could pre-initialize the wizard with the known username, but it's low priority detail.